### PR TITLE
fix: fetch timeout has better error UX

### PR DESCRIPTION
**Pull request by SgtPooki**
PR to address issue pointed out in explore.ipld.io PR comment: https://github.com/ipld/explore.ipld.io/pull/119#pullrequestreview-1519319888

image

### CODE REVIEW

File: src/lib/errors.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   * You must pass a t function that is registered with namespace 'explore' and keyPrefix 'errors'.
+   *
+   * @param t - the i18next-react t function
+   * @returns the translated string
+   * @example
+   * const {t} = useTranslation('explore', { keyPrefix: 'errors' })
+   * t('NameOfErrorClassThatExtendsIpldExploreError')
+   */
+  toString (t: TFunction<'translation', undefined, 'translation'>): string {
+    return t(this.constructor.name, this.options)
```
Comment (SgtPooki) [inline src/lib/errors.ts:20]: willfix

**Review by SgtPooki — COMMENTED** [review]
self review

**Comment by SgtPooki** [comment]
Note: if an error does not extend `IpldExploreError`, it's regular `toString()` result is output in the error message.

**Comment by lidel** [comment]
Awesome!  :heart:

**Pull request merged**